### PR TITLE
[2.0] Migrate user models to the new app

### PR DESCRIPTION
**Pull request by matti-lamppu**
# NOTE: Must merge #1339 SEPARATELY before this PR so that migrations work correctly!

## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- See title.
- Old migrations will remain in place until everything is moved to not break the django migration system.
- Should follow [this guide](https://realpython.com/move-django-model/#the-short-way-reference-the-new-django-model-to-the-old-table) (the short way).
- Note: User migrations must come before all other migrations so that migration dependencies can be put in the correct order.

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3547](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3547)

[TILA-3547]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3547?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by vergama — APPROVED** [review]

**Comment by matti-lamppu** [comment]
Merged #1341, #1342, #1343, #1344, #1345, #1346, #1347, #1348, #1363, #1364, #1365, #1367 & #1369 to this PR so that they can be merged to main together.

File: tilavarauspalvelu/models/reservation/actions.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+from django.utils.translation import pgettext_lazy
+from icalendar import Calendar, Event
+
+from common.date_utils import DEFAULT_TIMEZONE, local_datetime
+from common.utils import get_attr_by_language
+from tilavarauspalvelu.enums import CalendarProperty, EventProperty
+
+if TYPE_CHECKING:
+    from common.typing import Lang
+    from reservation_units.models import ReservationUnit
```
Comment (ranta) [inline tilavarauspalvelu/models/reservation/actions.py:15]: `ReservationUnit` is imported from the wrong app

File: tilavarauspalvelu/models/recurring_reservation/actions.py [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+)
+from tilavarauspalvelu.utils.opening_hours.time_span_element import TimeSpanElement
+
+if TYPE_CHECKING:
+    from collections.abc import Collection, Iterable
+
+    from django.db import models
+
+    from applications.models import City
+    from users.models import User
```
Comment (ranta) [inline tilavarauspalvelu/models/recurring_reservation/actions.py]: Imports from wrong apps

**Pull request merged**